nile-backend/src/main/java/com/tp/Nile/controllers/ProductController.java

```diff
@@ -1,8 +1,8 @@
 package com.tp.Nile.controllers;
 
 
-import com.tp.Nile.exceptions.InvaildProductIdException;
-import com.tp.Nile.exceptions.NullProductIdException;
+import com.tp.Nile.controllers.requests.AddProductRequest;
```

Comment:
I'm not totally sure but I thought with JPA we could just use the Product model as our RequestBody.

Reply:
With the Product model, you would have to enter all the associated info such as features and photos when you add a product. The AddProductRequest is a work around to only enter the ids of features and photos when you add a product

Reply:
there are downsides to JPA